parsing/parser.mly

```diff
@@ -2302,7 +2303,8 @@ simple_expr:
   | od=open_dot_declaration DOT LBRACE record_expr_content RBRACE
       { let (exten, fields) = $4 in
         (* TODO: review the location of Pexp_construct *)
-        Pexp_open(od, mkexp ~loc:$sloc (Pexp_record(fields, exten))) }
+        Pexp_open(od, mkexp ~loc:($startpos($3), $endpos)
```

Comment:
It would be nicer if instead of `($startpos($3), $endpos)` we could write this as the location of *something*, for example by having `od=open_dot_declaration DOT content=delimited(LBRACE, record_expr_content, RBRACE)`.